main_window/competition/mapping/mapping_thread.py

```diff
@@ -94,6 +95,12 @@ def setViewedDevice(self, devices) -> None:
         """
         with self.cv:
             self.viewed_devices = devices
+            for device in self.viewed_devices:
+                if self.device_status[device] == False: #no data for this device
```

Comment:
maybe there's a better way to track whether data is available instead of keeping this device_status dictionary. 

If data unavailable, there is a logger warning but no change on the map.

Reply:
You could totally just call do `self.rocket_data.last_value_by_device(device, DataEntryIds.LATITUDE) is None`. There's no downside to calling that function more times. Looks like it would save a few lines too.